productcat/src/main/java/com/visenze/productcat/android/util/ResponseParser.java

```diff
@@ -120,6 +122,73 @@ public static ResultList parseResult(String jsonResponse, String type) {
         }
     }
 
+    public static List<ObjectList> parseMPSResponse(JSONObject jsonObject) {
+        List<ObjectList> objectList = new ArrayList<ObjectList>();
+        try {
+            if (jsonObject.has(OBJECTS)) {
+                JSONArray objArray = jsonObject.getJSONArray(OBJECTS);
+                int size = objArray.length();
+                for (int i = 0; i < size; i++) {
+
+                    ObjectList obj = new ObjectList();
+                    if (jsonObject.has(REQID)) {
+                        obj.setReqid(jsonObject.getString(REQID));
+                    }
+
+                    JSONObject jsonItem = objArray.getJSONObject(i);
+
+                    if (jsonItem.has(TYPE)) {
+                        obj.setType((jsonItem.getString(TYPE)));
+                    }
+
+                    if (jsonItem.has(BOX)) {
+
+                        JSONArray boxCoordinate = jsonItem.getJSONArray(BOX);
+                        Box box = new Box(boxCoordinate.getInt(0),
```

Comment:
box might be empty, should add a check here e.g. if our API is called with `box` parameter, the box would be empty array.

Reply:
updated already